src/frontend/komponenter/Fagsak/Vilkårsvurdering/GeneriskVilkår/useAvslagBegrunnelseMultiselect.ts

```diff
@@ -9,25 +13,39 @@ const useAvslagBegrunnelseMultiselect = (vilkårType: VilkårType, regelverk?: R
     const { vedtaksbegrunnelseTekster } = useVedtaksbegrunnelseTekster();
 
     if (vedtaksbegrunnelseTekster.status !== RessursStatus.SUKSESS) {
-        return { avslagBegrunnelseTeksterForGjeldendeVilkår: [] };
+        return { grupperteAvslagsbegrunnelser: [] };
     }
 
-    const nasjonaleBegrunnelser = vedtaksbegrunnelseTekster.data.AVSLAG.filter(
-        (begrunnelse: IRestBegrunnelseTilknyttetVilkår) => begrunnelse.vilkår === vilkårType
-    );
+    const gyldigeBegrunnelseTyper = [BegrunnelseType.AVSLAG];
+
+    if (regelverk == Regelverk.EØS_FORORDNINGEN) {
+        gyldigeBegrunnelseTyper.push(BegrunnelseType.EØS_AVSLAG);
+    }
 
-    const avslagBegrunnelseTeksterForGjeldendeVilkår =
-        regelverk == Regelverk.EØS_FORORDNINGEN
-            ? nasjonaleBegrunnelser.concat(
-                  vedtaksbegrunnelseTekster.data.EØS_AVSLAG.filter(
-                      (begrunnelse: IRestBegrunnelseTilknyttetVilkår) =>
-                          begrunnelse.vilkår === vilkårType
-                  )
-              )
-            : nasjonaleBegrunnelser;
+    const grupperteBegrunnelser = Object.keys(vedtaksbegrunnelseTekster.data)
+        .filter((begrunnelseType: string) =>
+            gyldigeBegrunnelseTyper.includes(begrunnelseType as BegrunnelseType)
+        )
+        .reduce((acc: GroupBase<ISelectOption>[], begrunnelseType: string) => {
+            return [
+                ...acc,
+                {
+                    label: begrunnelseTyper[begrunnelseType as BegrunnelseType],
+                    options: vedtaksbegrunnelseTekster.data[begrunnelseType as BegrunnelseType]
+                        .filter(
+                            (begrunnelse: IRestBegrunnelseTilknyttetVilkår) =>
+                                begrunnelse.vilkår === vilkårType
+                        )
+                        .map((begrunnelse: IRestBegrunnelseTilknyttetVilkår) => ({
+                            label: begrunnelse.navn,
+                            value: begrunnelse.id,
+                        })),
+                },
+            ];
+        }, []);
```

Comment:
Da tror jeg også vi kan gjøre:
```suggestion
        .reduce((acc: GroupBase<ISelectOption>[], begrunnelseType) => {
            return [
                ...acc,
                {
                    label: begrunnelseTyper[begrunnelseType],
                    options: vedtaksbegrunnelseTekster.data[begrunnelseType]
                        .filter(
                            (begrunnelse: IRestBegrunnelseTilknyttetVilkår) =>
                                begrunnelse.vilkår === vilkårType
                        )
                        .map((begrunnelse: IRestBegrunnelseTilknyttetVilkår) => ({
                            label: begrunnelse.navn,
                            value: begrunnelse.id,
                        })),
                },
            ];
        }, []);
```